x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/AttributeMap.java

```diff
@@ -393,7 +379,10 @@ private Builder() {}
         }
 
         public AttributeMap<E> build() {
-            return new AttributeMap<>(map);
+            // copy, in case someone would do a .build, .put, .build sequence
+            AttributeMap<E> m = new AttributeMap<>();
+            m.addAll(map);
+            return m;
```

Comment:
Why not return the map directly? `AttributeMap` is immutable.
To guarantee that the builder is not used *after* `build()` is called, a flag could be used instead of copying things over.

Reply:
Changed, won't copy on `.build()`, only will copy once on the first `.put()` after a `.build()`.

Reply:
Removed the defensive copy completely: https://github.com/elastic/elasticsearch/commit/c4e3fc45f0a781d528c198866ee3de178f562991